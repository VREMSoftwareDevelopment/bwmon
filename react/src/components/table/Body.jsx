/*
 *      Copyright (C) 2010 - 2025 VREM Software Development <VREMSoftwareDevelopment@gmail.com>
 *
 *      Licensed under the Apache License, Version 2.0 (the "License");
 *      you may not use this file except in compliance with the License.
 *      You may obtain a copy of the License at
 *
 *           http: //www.apache.org/licenses/LICENSE-2.0
 *
 *      Unless required by applicable law or agreed to in writing, software
 *      distributed under the License is distributed on an "AS IS" BASIS,
 *      WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 *      See the License for the specific language governing permissions and
 *      limitations under the License.
 *
 * Bandwidth Monitor
 */

import React from 'react';
import PropTypes from 'prop-types';
import { TableBody, TableCell, TableRow } from '@mui/material';

const Body = ({ prefix, cellInfos, values }) => (
    <TableBody>
        {values.map((value, index) => (
            <TableRow
                key={value.id}
                id={prefix + '-data-' + index}
                data-testid={prefix + '-data-' + index}
                style={index % 2 ? { background: 'ghostwhite' } : { background: 'white' }}
            >
                {cellInfos.map((cellInfo) => {
                    return (
                        <TableCell key={cellInfo.id} align={cellInfo.align}>
                            {cellInfo.convert ? cellInfo.convert(value[cellInfo.id]) : value[cellInfo.id]}
                        </TableCell>
                    );
                })}
            </TableRow>
        ))}
    </TableBody>
);

Body.propTypes = {
    prefix: PropTypes.string.isRequired,
    cellInfos: PropTypes.arrayOf(
        PropTypes.shape({
            id: PropTypes.string.isRequired,
            align: PropTypes.string,
            convert: PropTypes.func,
        })
    ).isRequired,
    values: PropTypes.arrayOf(PropTypes.object).isRequired,
};

export default Body;
