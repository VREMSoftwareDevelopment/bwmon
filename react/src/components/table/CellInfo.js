/*
 *      Copyright (C) 2010 - 2020 VREM Software Development <VREMSoftwareDevelopment@gmail.com>
 *
 *      Licensed under the Apache License, Version 2.0 (the "License");
 *      you may not use this file except in compliance with the License.
 *      You may obtain a copy of the License at
 *
 *           http: //www.apache.org/licenses/LICENSE-2.0
 *
 *      Unless required by applicable law or agreed to in writing, software
 *      distributed under the License is distributed on an "AS IS" BASIS,
 *      WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 *      See the License for the specific language governing permissions and
 *      limitations under the License.
 *
 * Bandwidth Monitor
 */

export default class CellInfo {
    constructor(id, sortable, align, label, footer, convert) {
        this.id = id;
        this.sortable = sortable;
        this.align = align;
        this.label = label;
        this.footer = footer;
        this.convert = convert;
    }
}
