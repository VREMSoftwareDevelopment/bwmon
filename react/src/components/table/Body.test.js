/*
 *      Copyright (C) 2010 - 2023 VREM Software Development <VREMSoftwareDevelopment@gmail.com>
 *
 *      Licensed under the Apache License, Version 2.0 (the "License");
 *      you may not use this file except in compliance with the License.
 *      You may obtain a copy of the License at
 *
 *           http: //www.apache.org/licenses/LICENSE-2.0
 *
 *      Unless required by applicable law or agreed to in writing, software
 *      distributed under the License is distributed on an "AS IS" BASIS,
 *      WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 *      See the License for the specific language governing permissions and
 *      limitations under the License.
 *
 * Bandwidth Monitor
 */

import React from 'react';
import { create } from 'react-test-renderer';
import Body from './Body';
import CellInfo from './CellInfo';

describe('Body', () => {
    const convert = (value) => value / 1000;

    const cellInfos = [new CellInfo('IP', true, 'left', 'IP', false), new CellInfo('up', false, 'right', 'Down', true, convert)];

    const values = [
        { id: 1, IP: '192.168.1.14', up: 12345202809 },
        { id: 2, IP: '192.168.1.21', up: 123453907 },
        { id: 3, IP: '192.168.1.10', up: 5370488 },
    ];

    it('renders correctly', () => {
        const tree = create(<Body prefix="test" cellInfos={cellInfos} values={values} />).toJSON();
        expect(tree).toMatchSnapshot();
    });
});
