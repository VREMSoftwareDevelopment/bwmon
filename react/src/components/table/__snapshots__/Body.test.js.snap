// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Body renders correctly 1`] = `
<tbody
  className="MuiTableBody-root css-gmh7jj-MuiTableBody-root"
  role={null}
>
  <tr
    className="MuiTableRow-root css-1xtozoh-MuiTableRow-root"
    id="test-data-0"
    role={null}
    style={
      Object {
        "background": "white",
      }
    }
  >
    <td
      aria-sort={null}
      className="MuiTableCell-root MuiTableCell-body MuiTableCell-alignLeft MuiTableCell-sizeMedium css-19syksp-MuiTableCell-root"
    >
      192.168.1.14
    </td>
    <td
      aria-sort={null}
      className="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight MuiTableCell-sizeMedium css-3ssuu9-MuiTableCell-root"
    >
      12345202.809
    </td>
  </tr>
  <tr
    className="MuiTableRow-root css-1xtozoh-MuiTableRow-root"
    id="test-data-1"
    role={null}
    style={
      Object {
        "background": "ghostwhite",
      }
    }
  >
    <td
      aria-sort={null}
      className="MuiTableCell-root MuiTableCell-body MuiTableCell-alignLeft MuiTableCell-sizeMedium css-19syksp-MuiTableCell-root"
    >
      192.168.1.21
    </td>
    <td
      aria-sort={null}
      className="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight MuiTableCell-sizeMedium css-3ssuu9-MuiTableCell-root"
    >
      123453.907
    </td>
  </tr>
  <tr
    className="MuiTableRow-root css-1xtozoh-MuiTableRow-root"
    id="test-data-2"
    role={null}
    style={
      Object {
        "background": "white",
      }
    }
  >
    <td
      aria-sort={null}
      className="MuiTableCell-root MuiTableCell-body MuiTableCell-alignLeft MuiTableCell-sizeMedium css-19syksp-MuiTableCell-root"
    >
      192.168.1.10
    </td>
    <td
      aria-sort={null}
      className="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight MuiTableCell-sizeMedium css-3ssuu9-MuiTableCell-root"
    >
      5370.488
    </td>
  </tr>
</tbody>
`;
